File: Data Structures & Algorithms/is-anagram/submission-4.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        #include <unordered_map>

        unordered_map <char, int> hashmap;

        if ( s.length() < t.length() || t.length() < s.length())
        {
            return false;
        }
        for (int i = 0; i < s.length(); i++)
        {
            if (hashmap.find(s[i]) == hashmap.end())
            {
                hashmap[s[i]] = 1;
            }
            else
            {
                hashmap[s[i]] += 1;
            }
        }
        for (int i = 0; i < t.length(); i++)
        {
            if (hashmap.find(t[i]) == hashmap.end() or hashmap[t[i]] == 0)
            {
                return false;
            }
            else
            {
                hashmap[t[i]] -= 1;
            }
        }

        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        new_arr = []

        for i in range(len(nums)):
            hashmap[nums[i]] = 1 + hashmap.get(nums[i], 0)
        
        for n in range(k):
            highestKey = 0
            highestVal = 0
            for key in hashmap:
                if hashmap[key] > highestVal:
                    highestVal = hashmap[key]
                    highestKey = key
            new_arr.append(highestKey)
            del hashmap[highestKey]

        return new_arr